lib/ts/recipe/session/sessionClass.ts

```diff
@@ -55,12 +55,17 @@ export default class Session implements SessionContainerInterface {
 
     getSessionData = async (userContext?: any): Promise<any> => {
         try {
-            return (
-                await this.helpers.sessionRecipeImpl.getSessionInformation({
-                    sessionHandle: this.sessionHandle,
-                    userContext: userContext === undefined ? {} : userContext,
-                })
-            ).sessionData;
+            let sessionInfo = await this.helpers.sessionRecipeImpl.getSessionInformation({
+                sessionHandle: this.sessionHandle,
+                userContext: userContext === undefined ? {} : userContext,
+            });
+            if (sessionInfo === undefined) {
+                throw new STError({
```

Comment:
This is caught a few lines below, I'd prefer if we just did the cleanup here without a try-catch + rethrow. This also applies to: `updateSessionData`, `updateAccessTokenPayload`, `getTimeCreated`, and `getExpiry`.

Reply:
Fair point. Done.